### PR TITLE
Support Meteor 3 (without fibers)

### PR DESCRIPTION
**Pull request by leonardoventurini**

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+export const isFibersInstalled = isInstalled('fibers');
```
Comment (zodern) [inline src/utils.js]: We should be able to use `Meteor.isFibersDisabled`.

File: test/server.js [hunk 2]
```diff
@@ -1,3 +1,5 @@
+new Mongo.Collection('dummy');
```
Comment (zodern) [inline test/server.js]: Why is this needed? It might be good to add a comment on why an unused collection is created here.

File: package.js [hunk 3]
```diff
@@ -17,7 +17,12 @@ Package.onTest(function(api) {
 });
 
 function configurePackage(api) {
-  api.versionsFrom("METEOR@1.0");
-  api.use(["random", "mongo"], "server");
+  api.versionsFrom("METEOR@2.0");
```
Comment (zodern) [inline package.js]: This probably should still be METEOR@1.0 to avoid breaking changes, but adding a dependency on `ecmascript` prevents that. Changing the minimum version to 1.4 would probably be good enough.

File: package.js [hunk 4]
```diff
@@ -1,6 +1,6 @@
 Package.describe({
   summary: "Exposes internal Meteor APIs",
-  version: "2.2.0",
+  version: "3.0.0",
```
Comment (zodern) [inline package.js]: We want to first release a new 2.x version without any breaking changes (I think the only one intended was making the `mongo` package optional).

File: src/server.js [hunk 5]
```diff
@@ -13,25 +22,31 @@ MeteorX.onReady = function(cb) {
...
-  if (Fibers.current) {
-    cb();
-  } else {
-    new Fiber(cb).run();
+  for (const cb of MeteorX._readyCallbacks) {
+    await cb();
   }
+
+  MeteorX._ready = true;
 }
```
Comment (zodern) [inline src/server.js:44]: I'm wondering if we can combine these two functions without adding much complexity:
```js
async function init() {
  if (fibersEnabled) {
    exposeMongoLivedata(MeteorX);
  } else {
    await exposeMongoAsync(MeteorX);
  }

 for (const cb of MeteorX._readyCallbacks) {
     await cb();
  }

  MeteorX._ready = true;
}
```

Before Meteor 3, async functions in packages/app code always run within a fiber so we don't have to manually wrap it ourselves

If we want to make startup wait for this async function, we could do something like this:
```
Meteor.startup(() => {
  let promise = init();
  if (promise.await) promise.await();

  return promise;
});
```

File: src/server.js [hunk 6]
```diff
@@ -1,8 +1,17 @@
...
+import { isFibersInstalled, isMongoInstalled, runWithAFiber, wrapFn } from "./utils";
+import { exposeMongoAsync } from "./fiberless/mongo";
+import { installMongoDetector } from "./mongo-detector";
 
+/**
+ * @namespace MeteorX
+ */
 MeteorX = {};
+
+MeteorX._hasInitializedMongo = false;
```
Comment (zodern) [inline src/server.js]:
```suggestion
MeteorX._mongoInitialized = false;
```

Minor suggestion. I think this would make some of the logic using this boolean easier to understand.

File: src/server.js [hunk 7]
```diff
@@ -13,25 +22,31 @@ MeteorX.onReady = function(cb) {
...
-  MeteorX._readyCallbacks.forEach(function(fn) {
-    runWithAFiber(fn);
-  });
+  MeteorX._readyCallbacks.map(runWithAFiber);
   MeteorX._ready = true;
-});
+}
+
+async function initAsync() {
+  await exposeMongoAsync(MeteorX);
```
Comment (zodern) [inline src/server.js:37]: This is called twice: once in `installMongoDetector`, and again here. This second call doesn't seem to do anything.

The goal was to only expose the classes from the mongo package if the app connected to MongoDB.

The current solution has two issues:
1. It fails to detect it if all collections are created before meteorx loads (the app doesn't use any collections, but has packages that do. This could happen in apps that have a package based architecture).
2. there is no guarantee that a collection will be created before `initAsync` is called - it could happen later, after the ready callbacks are run.

For the first one, there is no great option - we are going to have false negatives or false positives. I'm not sure what the right solution is. Another idea is to check if the app is using the mongo package, and the MONGO_URL env var is defined and then expose mongo. The main downside is if for some reason an app doesn't create any mongo collections, but loads the mongo package and has an old and broken MONGO_URL, it would cause the app to crash. Meteorx already has that issue and so far we haven't received any reports about it.

For the second issue, we might want to add a separate callback for mongo or some other way or something similar, or use the previous suggestion which would cause it to always be exposed before the ready callbacks are run.

However, we want to first release a version without breaking changes, so it should always expose mongo like meteorx currently does. This would be relevant for the first version 3 release.

File: test/server.js (hunk 2 above)
Comment (leonardoventurini) [reply test/server.js]: I think that was to force mongo to initialize, no longer needed I removed the mongo detector for now.

File: src/server.js (hunk 5 above)
Comment (leonardoventurini) [reply src/server.js:44]: I don't think it is a good idea to combine them, having them separate makes it way easier to understand and maintain, even if we have some duplicated code.

File: src/server.js (hunk 7 above)
Comment (leonardoventurini) [reply src/server.js:37]: I removed the optional mongo behavior, saved this comment for later.
